src/app/home-page/recent-events/recent-events.component.ts

```diff
@@ -57,15 +98,48 @@ export class RecentEventsComponent implements OnInit {
           this.alertService.detailedError(error);
         }
       );
+      this.getAUsersEvents();
+      this.loading$ = this.recentEventsQuery.selectLoading();
     } else {
-      this.recentEventsService.get();
+      // On dashboard, get my events
+      this.isLoading = true;
+      this.supportedEventTypes = this.eventType === 'SELF_ORGANIZATIONS' ? this.supportedOrgEventTypes : this.supportedStarredEventTypes;
+      this.getMyEventsByType();
     }
+  }
+
+  /**
+   * Get another user's events when viewing their user page
+   */
+  private getAUsersEvents() {
+    this.recentEventsQuery.selectAll({ filterBy: (entity) => this.supportedUserEventTypes.includes(entity.type) }).subscribe(
+      (events) => {
+        this.events = events;
+      },
+      (error: HttpErrorResponse) => {
```

Comment:
This is querying the Akita store, not making an http(s) request. It looks like we generally don't need to handle the error condition for querying the store, since it's just fetching from an in memory object.

Also, you're supposed to unsubscribe. The preferred way is to just return an observable and use `async` in HTML, but I can see how that might be tricky because you're also potentially setting this property from an https request. The other way to do it would be with `pipe` and `takeUntil(this.ngUnsubscribe)`.

The third way to so save the subscription and unsubscribe, but that one's a hassle :)